iroh-net/src/discovery/dns.rs

```diff
@@ -57,25 +54,13 @@ impl DnsDiscovery {
     ///
     /// # Usage during tests
     ///
-    /// When `cfg(test)` is enabled or when using the `test-utils` cargo feature the
-    /// [`TEST_DNS_NODE_ORIGIN`] is used.
-    ///
-    /// Note that the `iroh.test` domain is not integrated with the global DNS network and
-    /// thus node discovery is effectively disabled.  To use node discovery in a test use
-    /// the [`crate::test_utils::DnsPkarrServer`] in the test and configure it as a
-    /// custom discovery mechanism.
-    ///
-    /// For testing it is also possible to use the [`N0_DNS_NODE_ORIGIN_STAGING`] domain
-    /// with [`DnsDiscovery::new`].  This would then use a hosted discovery service again,
-    /// but for testing purposes.
+    /// For testing it is possible to use the [`N0_DNS_NODE_ORIGIN_STAGING`] domain
+    /// with [`DnsDiscovery::new`].  This would then use a hosted staging discovery
+    /// service for testing purposes.
     pub fn n0_dns() -> Self {
-        #[cfg(not(any(test, feature = "test-utils")))]
-        {
-            Self::new(N0_DNS_NODE_ORIGIN_PROD.to_string())
-        }
-        #[cfg(any(test, feature = "test-utils"))]
-        {
-            Self::new(TEST_DNS_NODE_ORIGIN.to_string())
+        match force_staging_infra() {
+            true => Self::new(N0_DNS_NODE_ORIGIN_STAGING.to_string()),
+            false => Self::new(N0_DNS_NODE_ORIGIN_PROD.to_string()),
```

Comment:
```suggestion
        if force_staging_infra() {
            Self::new(N0_DNS_NODE_ORIGIN_STAGING.to_string())
        } else {
            Self::new(N0_DNS_NODE_ORIGIN_PROD.to_string())
```

Reply:
I guess it'd be unusual to see a match on booleans. But honestly, it's shorter, and I'd almost say the match reads better in this case? I'm somewhat undecided :D

Reply:
Yeah, I just followed like the others (where I `let x = match ...`) but technically here it is a simple if else. Updated to that 🤷

Reply:
lol, I saw the other uses and was like "ah well, it's probably fine", but now we have it inconsistent :see_no_evil: 
Anyhow, we can update another time.